main.go

```diff
@@ -197,16 +200,25 @@ func main() {
 		// ``MINIO_PROFILER`` supported options are [cpu, mem, block].
 		switch os.Getenv("MINIO_PROFILER") {
```

Comment:
If the profiling switch statement is moved out of `app.Before` func and placed before `app.RunandExitOnError()` is called, then the profile output files would be placed in the default configuration directory as opposed to the configuration directory supplied to `minio` as a command line flag.